src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -316,6 +350,13 @@ private async Task<(string sourceBranch, string sourceVersion)> GetSourceBranchI
             if (hasSourceBranch)
             {
                 _options.SourceBranch = GitHelpers.NormalizeBranchName(_options.SourceBranch);
+
+                if (_options.SourceBranch.EndsWith("release/3.x", StringComparison.OrdinalIgnoreCase))
+                {
+                    Console.WriteLine($"Warning: Arcade branch {_options.SourceBranch} doesn't support build promotion. Please try specifiying --source-branch 'master'.");
+                    Console.WriteLine("Switching source branch to Arcade master.");
```

Comment:
So If I understand this correctly, we tell the user to consider switching to using master, but we switch for them anyways?

Reply:
Yes, that's correct.

Reply:
We should probably just go ahead and inform them that we're doing it. I don't see much value in telling them to do something that is getting handled automatically.